infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -864,4 +875,20 @@ private void updateReplicationStatus() {
             statusFlag = true;
         }
     }
+
+    private void setupLocalNodeId() {
+        // Retrieve system-specific node id
+        LogReplicationPluginConfig config = new LogReplicationPluginConfig(serverContext.getPluginConfigFilePath());
+        String nodeIdFilePath = config.getNodeIdFilePath();
+        if (nodeIdFilePath != null) {
+            File nodeIdFile = new File(nodeIdFilePath);
+            try (BufferedReader bufferedReader = new BufferedReader(new FileReader(nodeIdFile))) {
+                String line = bufferedReader.readLine();
+                localNodeId = Optional.of(line.split("=")[1].trim().toLowerCase());
+                log.info("setupLocalNodeId succeeded, node id is {}", localNodeId);
+            } catch (Exception e) {
+                log.warn("setupLocalNodeId failed", e);
+            }
+        }
```

Comment:
can we add a logging warning if no Node Id file path is found?

Reply:
Added